src/features/surveys/components/SurveySubmissionsList.tsx

```diff
@@ -198,6 +208,12 @@ const SurveySubmissionsList = ({
         id: row.id,
       });
       setRespondentId(person?.id || null);
+
+      const personHasNoEmail = person?.id !== null && person?.email === null;
+      const respondentEmail = row.respondent?.email;
+      if (personHasNoEmail && respondentEmail !== undefined) {
```

Comment:
Whenever I see strict equality checks (`===` and `!==`) in typescript, that is a bit of a yellow flag for me.

### Why strict equality?
The purpose of the strict equality operators in javascript (as opposed to  the regular equality operators, `==` and `!=`) is to compare the values without making any type conversions. The regular operator is "smart" in that it tries to convert the operands to types that can be compared, but in doing so it sometimes does the wrong thing. For example, `null == undefined` is true (because one or both operands are converted to a compatible type), but `null === undefined` is not (because they are different types). Similarly, `1 == '1'` is true, because the left operand is converted to a string before comparing, but `1 === '1'` is not true, because the left operand is a number and the right is a string.

The strict equality is important in javascript, because types are so dynamic that you're encouraged not to care about what type something will be at execution time. Using strict equality is a good default in javascript, because it lets you know (by not working as intended) if something is of a different type than you expected. 

For example, imagine this JS code:

```js
if (a == 5) {
  const eight = a + 3;
  console.log(eight);
}
```

Will it print 8? We can't know! Because if `a` is `'5'` (the string), the boolean expression will evaluate to true (because `'5' == 5`) but `eight` will actually be `'53'` (the string resulting from concatenating the two strings '5' and '3').

In this case, the "cleverness" of the default equality operator in JS is biting us. If we had used strict equality (`===`) the if statement would not have executed.

That is probably _also not what we want_, but at least it's "jarring" enough for the tester/user to hopefully discover that there is some problem here, and the programmer can go and find the original flaw, i.e. whatever is causing `a` to be a string when the code assumes it's a number.

That's why using `===` in javascript is a good idea. It helps you find bugs related to types, but it doesn't fix them.

### Strict equality and Typescript
Typescript, however, is different. Although typescript is of course only javascript when it gets executed, the problem of not knowing types is (mostly) avoided by compile-time type-checking.

Typescript type-checks operands in equality expressions. This means that if the two operands have different types, typescript will fail to compile. You can [see this in action here](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBAhjAvDA5AVhQbgFAEsBmMAFAosmgJQwDe2MMoksAprgOYAWsyCA1DAGYc9RhBAAbZgDpxINkVacoFHAF9s2UbABGSGGhwFiusvqq0R4aDEVc9u-kLoMrE6bPm3laoA):

![image](https://github.com/user-attachments/assets/c0edf8f4-7b02-4615-896e-34c1233eb93f)

The bugs that strict equality helps us find _at runtime_ (if we're lucky), typescript helps us find _at compile-time_ (always!).

This does not mean that strict equality never has a place in typescript. There are cases where typescript will allow a comparison between two separate (but related) types. For example, the expression `null == undefined` is valid in typescript and once executed in javascript will evaluate to true. So if you really want to distinguish `null` from `undefined`, you need strict equality.

Which leads us to your code.

### The problem(?) with your code
Your code is comparing `respondentEmail` with `undefined` using strict equality. This means that the if statement will only execute if `respondentEmail` is anything other than strictly `undefined`.

If `respondentEmail` is `null`, I would argue (in this context) that it's conceptually equal to `undefined`. The purpose of the code is to express some intent. In this case, the intent is to show a dialog if the respondent _has no email_, which at least in theory could mean that it's either `undefined` or `null`.

Now, I don't think that in this specific case `respondentEmail` can ever be `null`, because `resondent.email` will always be a string if it exists, so there is no bug here.

But if we ever make changes in the future to the type of `ZetkinSurveySubmission`, to allow `respondent.email` to be `null | string`, a bug would be introduced in this completely separate part of the code. Because even if we change the type, the _intent_ is still to open this dialog if the respondent _has no email_, but that won't happen because the email is not strictly `undefined`, which is what you're checking for here.

So, in conclusion, this code is being unnecessarily strict, probably due to an instinct conditioned from years of javascript programming, but you're not getting any of the benefits from that instinct, only the (potential) problems.

Reply:
Thank you so much for this detailed explanation and thoughtful feedback! I genuinely appreciate the time you took to not only review the code but also provide a deep dive into the reasoning behind strict equality checks in both JavaScript and TypeScript.
It’s clear that my instinct for writing `===` here came from JavaScript habits, but, as you said, the TypeScript compiler already ensures type safety. I’ll update the code and I will keep this lesson in mind for the future. Your mentorship is so great! 🙌